### PR TITLE
JS: Basic syntax done

### PR DESCRIPTION
**Pull request by mohammadkazemy**
Closes: #60

### CODE REVIEW

File: mohammad/JS/JS-basicSyntax/main.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+// flip a coin and log the result (head or tail)
+function flipACoin() {
+  if (Math.random() < 0.5) console.log("head");
+  else console.log("tail");
+}
+
+// concat two strings
+function concat(stringA, stringB) {
+  return stringA + stringB;
```
Comment (Kiarash-ip) [inline mohammad/JS/JS-basicSyntax/main.js:14]: this is ok, but there is another approach for concating strings which is used more

Comment (mohammadkazemy) [reply mohammad/JS/JS-basicSyntax/main.js:14]: Can you tell me what that one is?

File: mohammad/JS/JS-basicSyntax/main.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+// higher order function: call the function in the parameters list
+// e.g. callFuncNTimes(flipACoin, 3) calls flipACoin 3 times
+function callFuncNTimes(func, n) {
+  for (let i = 0; i < n; i++) func();
+}
+
+// return true if parameter is falsy, false otherwise
+function isNil(param) {
+  if (param == false) return true;
```
Comment (Kiarash-ip) [inline mohammad/JS/JS-basicSyntax/main.js:79]: if statement converts its condition to truthy, so you don't have to compare it to boolean value

File: mohammad/JS/JS-basicSyntax/main.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  else if (param == true) return false;
+}
+
+// call property in object
+// e.g. in obj = { callableProp: function () { console.log('hello!') } }, callPropInObj(obj, 'callableProp') will log "hello!"
+function callPropInObj(object, propertyName) {
+  //return object["propertyName"];
+  return object.propertyName;
```
Comment (Kiarash-ip) [inline mohammad/JS/JS-basicSyntax/main.js:87]: it doesn't call the specified property

File: mohammad/JS/JS-basicSyntax/main.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+// if the email is gmail
+function isGmail(email) {
+  let condition = email.slice(-1, 9);
```
Comment (Kiarash-ip) [inline mohammad/JS/JS-basicSyntax/main.js:92]: If you want to cut the string from the end, you are giving the wrong value to the method

File: mohammad/JS/JS-basicSyntax/main.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+  if (condition == "gmail.com") return true;
+  return false;
+}
+
+// duplicate array
+// e.g. duplicateArray([1,2,3]) is [1,2,3,1,2,3]
+function duplicateArray(arr) {
+  return arr.append(arr);
```
Comment (Kiarash-ip) [inline mohammad/JS/JS-basicSyntax/main.js:100]: I am sure that this method does not work for JavaScript arrays

**Review by Kiarash-ip — CHANGES_REQUESTED** [review]

**Review by Kiarash-ip — DISMISSED** [review]

File: mohammad/JS/JS-basicSyntax/main.js [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+function getNRandomNumbers(n) {
+  let arr = [];
+  for (let i = 0; i < n; i++) arr[i] = Math.random();
+  return arr;
+}
+
+// get sorted array of n random numbers, in range min to max
+function getNRandomNumbers(n, min, max) {
+  let arr = [];
+  for (let i = 0; i < n; i++) arr[i] = Math.random() * (max - min) + min;
```
Comment (mkermani144) [inline mohammad/JS/JS-basicSyntax/main.js:50]: This doesn't seem sorted.

**Review by mkermani144 — CHANGES_REQUESTED** [review]

**Review by mkermani144 — APPROVED** [review]

**Pull request merged**